core/utils/string.hpp

```diff
@@ -375,143 +172,66 @@ inline size_t common_prefix_length(const Char* lhs, size_t lhs_size,
 }
 
 template<typename Char, typename Traits>
-inline size_t common_prefix_length(
-  const basic_string_ref<Char, Traits>& lhs,
-  const basic_string_ref<Char, Traits>& rhs) noexcept {
-  return common_prefix_length(lhs.c_str(), lhs.size(), rhs.c_str(), rhs.size());
-}
-
-template<typename T, typename U>
-inline void assign(std::basic_string<T>& str, const basic_string_ref<U>& ref) {
-  static_assert(sizeof(T) == 1 && sizeof(T) == sizeof(U));
-  str.assign(reinterpret_cast<const T*>(ref.c_str()), ref.size());
-}
-
-template<typename _Elem, typename _Traits>
-inline constexpr bool starts_with(const basic_string_ref<_Elem, _Traits>& first,
-                                  const _Elem* second) {
-  return starts_with(first, second, _Traits::length(second));
-}
-
-template<typename Elem, typename Traits>
-inline bool starts_with(const basic_string_ref<Elem, Traits>& first,
-                        const std::basic_string<Elem>& second) {
-  return starts_with(first, second.c_str(), second.size());
+inline size_t CommonPrefixLength(
+  std::basic_string_view<Char, Traits> lhs,
+  std::basic_string_view<Char, Traits> rhs) noexcept {
+  return CommonPrefixLength(lhs.data(), lhs.size(), rhs.data(), rhs.size());
 }
 
-template<typename _Elem, typename _Traits>
-inline constexpr bool starts_with(
-  const basic_string_ref<_Elem, _Traits>& first,
-  const basic_string_ref<_Elem, _Traits>& second) {
-  return starts_with(first, second.c_str(), second.size());
-}
-
-typedef basic_string_ref<char> string_ref;
-typedef basic_string_ref<byte_type> bytes_ref;
+template<typename Char>
+inline std::basic_string_view<Char> kEmptyStringView{
+  reinterpret_cast<const Char*>(""), 0};
```

Comment:
`L""` not same in bytes as `""`

Probably will be ok something like:
```cpp
inline constexpr char const* nulls = "\0\0\0";
```
https://godbolt.org/z/6xce3TYoE